examples/Microsoft.Spark.CSharp.Examples/Sql/Batch/VectorUdfs.cs

```diff
@@ -56,39 +58,28 @@ public void Run(string[] args)
             spark.Stop();
         }
 
-        private static RecordBatch CountCharacters(
-            RecordBatch records,
+        private static FxDataFrame CountCharacters(
+            FxDataFrame dataFrame,
             string groupFieldName,
             string stringFieldName)
         {
-            int stringFieldIndex = records.Schema.GetFieldIndex(stringFieldName);
-            StringArray stringValues = records.Column(stringFieldIndex) as StringArray;
-
             int characterCount = 0;
 
-            for (int i = 0; i < stringValues.Length; ++i)
+            PrimitiveDataFrameColumn<int> characterCountColumn = new PrimitiveDataFrameColumn<int>(stringFieldName + "CharCount");
+            PrimitiveDataFrameColumn<int> ageColumn = new PrimitiveDataFrameColumn<int>(groupFieldName);
+            for (long i = 0; i < dataFrame.RowCount; i++)
             {
-                string current = stringValues.GetString(i);
-                characterCount += current.Length;
+                characterCount += ((string)dataFrame[stringFieldName][i]).Length;
```

Comment:
Can you cache `dataFrame[stringFieldName]` into a variable, and then you don't need to cast in the loop.